jax/dtypes.py

```diff
@@ -77,6 +77,8 @@ class _bfloat16_finfo(object):
 @util.memoize
 def canonicalize_dtype(dtype):
   """Convert from a dtype to a canonical dtype based on FLAGS.jax_enable_x64."""
+  if isinstance(dtype, str) and dtype == "bfloat16":
+    dtype = bfloat16
```

Comment:
What's changed that makes this needed?

Reply:
Passing all float dtypes to ``testRngUniform`` leads to an error for bfloat16 without this change.